kommandir/bin/adept_openstack.py

```diff
@@ -1357,18 +1357,17 @@ def activate_and_setup(namespace, venvdir, requirements, onlybin, nobin):
 # N/B: Any/All names used here are in the global scope
 if __name__ == '__main__':  # pylint: disable=C0103
     # Parse arguments
-    basename = os.path.basename(sys.argv[0])
-    if basename == DISCOVER_CREATE_NAME:
-        _dargs = parse_args(sys.argv, 'discover')
-    elif basename == ONLY_CREATE_NAME:
-        _dargs = parse_args(sys.argv, 'exclusive')
-    elif basename == DESTROY_NAME:
-        _dargs = parse_args(sys.argv, 'destroy')
-    elif basename == REAP_NAME:
-        _dargs = parse_args(sys.argv, 'reap')
-    else:
-        logging.error("Script was not called as %s, %s, or %s", *ALLOWED_NAMES)
+    basename = re.sub(r'\.py$', '', os.path.basename(sys.argv[0]))
+    op_requested = re.sub('^openstack_', '', basename)
```

Comment:
IDK in this instance, but for future ref: ``re`` is much more "expensive" than the simpler string manip. functions.

Reply:
Python people are neurotic about re. Shrug. This is about maintainability.

Reply:
Hmmm...not sure I agree...I think it's just me that's neurotic :smile:

Reply:
gaah...github cross-comment-post.  What I mean is:  using 're' here is fine, we don't care about performance in this spot.   And you're right, it's more maintainable and less cryptic.

Reply:
Ha! No, I see it all the time when looking up python answers on StackOverflow. "re is inefficient", "re is unreadable", "re is evil". It's really weird, and counterproductive. I do agree that the Python re implementation is crap, as is pretty much all Python string handling, but the extent of the loathing for re is striking.

Reply:
> extent of the loathing for re is striking.

Interesting, I haven't encountered that.  Though I think you'll find such rantings about any language/system...people have opinions and love to spread them, especially when you disagree :smile:  But really the context definitely matters, for example right here, maintainability is _way_ more important than performance.  Best guess, it's got it's roots in 'C' (an original impetus and model for python). OTOH, try doing regex or printf stuff in the kernel, or for microcontroller and you'll be beaten to death with your own ripped-off limbs, then thrown into a volcano :laughing: